### ISSUE TITLE
Add checks that support wildcard specifications

### ISSUE THREAD

**Issue by gkapfham** [GatorEducator/gatorgrader#98]
The current implementation of GatorGrader allows you to specify a check for a specific file. For instance, you can ask GatorGrader to ensure that a specific Java file has a certain number of single-line comments. However, it is not possible right now to specify checks that have "wildcards" for the matching files. If this feature existed, then it would be possible to have a check to ensure that **all** Java files in a directory have a specific number of single-line comments. A simpler example of this type of check might involve ensuring that there exist a specific number of Java files inside of a directory.

Although I have not completely thought through all of the implementation details for this feature, I anticipate that it would be possible to implement it by using the `pathlib` library. Could someone please investigate further the feasibility of this feature? This would be particularly useful for final projects in which the assignment's specification might require a directory to contain a certain number of, say, Python or Java files.

**Comment by miller365** [comment]
So after some initial research, there is a way of doing something like this in python using a module called [fnmatch](https://docs.python.org/3/library/fnmatch.html). I think I could handle implementing this if I could get some guidance on how you would want it integrated into the current codebase. Regardless I will start working on this issue.

**Comment by gkapfham** [comment]
Also, @miller365 can you clarify which account you are going to use for contributions? Is is @miller365 or @cmiller365? Since @cmiller365 is already in the GitHub commit log, could I trouble you to add a photograph or an icon to your account so that people can know who you are and talk to you in person about your contributions? Thanks!

**Comment by gkapfham** [comment]
Hello @Michionlion and @schultzh and @corlettim, I would like to start a discussion about this issue! Right now, you have to specify checks in the following fashion:

```
termfrequency:
        compute_tf_aspects.py:
            --fragment "TODO" --count 0
```

However, if a system contains a total of 10 files, then you would have to write the above check for each one of the files, thereby ensuring that there are no `TODO` fragments inside of the code. I'm wondering if it would instead be acceptable to write the following code in the configuration file:

```
termfrequency:
        *.py:
            --fragment "TODO" --count 0
```

Notably, this makes it a lot easier to write a configuration file. With that said, it kinda violates our current assumption that a single run of GatorGrader corresponds to checks for a single file. What approach do you prefer? Are there other options that we have not yet considered? Please let me know. If we can coalesce on a final idea, then I might try to implement it later this week. Thanks!

**Comment by corlettim** [comment]
> Hello @Michionlion and @schultzh and @corlettim, I would like to start a discussion about this issue! Right now, you have to specify checks in the following fashion:
> 
> ```
> termfrequency:
>         compute_tf_aspects.py:
>             --fragment "TODO" --count 0
> ```
> 
> However, if a system contains a total of 10 files, then you would have to write the above check for each one of the files, thereby ensuring that there are no `TODO` fragments inside of the code. I'm wondering if it would instead be acceptable to write the following code in the configuration file:
> 
> ```
> termfrequency:
>         *.py:
>             --fragment "TODO" --count 0
> ```
> 
> Notably, this makes it a lot easier to write a configuration file. With that said, it kinda violates our current assumption that a single run of GatorGrader corresponds to checks for a single file. What approach do you prefer? Are there other options that we have not yet considered? Please let me know. If we can coalesce on a final idea, then I might try to implement it later this week. Thanks!

@gkapfham I prefer the idea of writing the code including the `.py` because I feel that writing one for each individual file could get very monotonous. I also think it would give a "comfort feel" to our tool for users...in other words, they can feel confident that all of their files are being checked rather than accidentally missing one or unsure of whether they will be needing a a specific check for an additional file down the line. However, you are right about that fact that it will violate the idea that GatorGrader runs checks for a single file...I feel like we can change the way this is worded to make it sound better and follow the idea that it checks all files while also making sure we do not underestimate the precise detail of checking that GatorGrade can do.

**Comment by Michionlion** [comment]
I think that the idea of atomic checks is more about a single purpose being followed (checking if a fragment exists, checking if a command runs, etc), and not necessarily that a single file is being acted upon. I think it's acceptable to add this, especially when we switch to the linting interface.

**Comment by gkapfham** [comment]
Hello @miller365 and @Michionlion and @corlettim, I have implemented this issue in PR #180.
